### ISSUE TITLE
Duplicate key for property path

### ISSUE THREAD

**Issue by inakie** [daaru00/gridsome-plugin-i18n#11]
**Describe the bug**

Thanks a lot for this plugin, it looks very promising!

I get the error "Duplicate key for property path" when trying to build. I am using the plugin @gridsome/source-filesystem to read the posts for a blog (.md files) and I believe that is where the problem is coming from.

**Plugin configuration**
```js
// lines with "es" locale commented to try to find the bug, but not working either

plugins: [
    {
      use: "gridsome-plugin-i18n",
      options: {
        locales: [
          // locales list
          "en-us",
          // "es-es",
        ],
        pathAliases: {
          // path segment alias for each locales
          "en-us": "en",
          // "es-es": "es",
        },
        // enablePathRewrite: true,
        fallbackLocale: "en-us", // fallback language
        defaultLocale: "en-us", // default language
        messages: {
          "en-us": require("./src/locales/en-us.json"), // Messages files
          // "es-es": require("./src/locales/es-es.json"),
        },
      },
    },
    {
      use: "@gridsome/source-filesystem",
      options: {
        baseDir: "./blog",
        pathPrefix: "/blog",
        path: "*.md",
        typeName: "BlogPost",
      },
    },
  ],
  templates: {
    BlogPost: "/blog/:id",
  },
```

**Output**
```
$ gridsome build
Gridsome v0.7.14

Initializing plugins...
Load sources - 0.07s
Create GraphQL schema - 0.06s
Error: Duplicate key for property path: /blog/8d665038979865f0a622d11ead24e5f6
    at UniqueIndex.set (/Users/inaki/dev/akademia/web/node_modules/lokijs/src/lokijs.js:7329:17)
    at Collection.add (/Users/inaki/dev/akademia/web/node_modules/lokijs/src/lokijs.js:5974:31)
    at Collection.insertOne (/Users/inaki/dev/akademia/web/node_modules/lokijs/src/lokijs.js:5738:17)
    at Collection.insert (/Users/inaki/dev/akademia/web/node_modules/lokijs/src/lokijs.js:5677:21)
    at Route.addPage (/Users/inaki/dev/akademia/web/node_modules/gridsome/lib/pages/pages.js:473:19)
    at Pages.createPage (/Users/inaki/dev/akademia/web/node_modules/gridsome/lib/pages/pages.js:176:18)
    at createPage (/Users/inaki/dev/akademia/web/node_modules/gridsome/lib/app/actions.js:260:24)
    at VueI18n.createManagedPages (/Users/inaki/dev/akademia/web/node_modules/gridsome-plugin-i18n/gridsome.server.js:63:7)
    at Plugins.run (/Users/inaki/dev/akademia/web/node_modules/gridsome/lib/app/Plugins.js:141:17)
    at process.internalTickCallback (internal/process/next_tick.js:77:7)
```

**Comment by borisdiakur** [comment]
I don’t think this has anything to do with @gridsome/source-filesystem.

I’ve set up a reproduction repository for this bug here: https://github.com/borisdiakur/repro-repo-gridsome-plugin-i18n-duplicate-key-bug
It is basically the bare bones [gridsome-starter-default](https://github.com/gridsome/gridsome-starter-default) with changes to [gridsome.server.js](https://github.com/borisdiakur/repro-repo-gridsome-plugin-i18n-duplicate-key-bug/blob/master/gridsome.server.js), where I set up a collection which uses [a new template](https://github.com/borisdiakur/repro-repo-gridsome-plugin-i18n-duplicate-key-bug/blob/master/src/templates/Product.vue) in the templates directory. When running the project with `yarn develop` without adding gridsome-plugin-i18n (you can comment out the plugin) you can navigate from [the start page](https://github.com/borisdiakur/repro-repo-gridsome-plugin-i18n-duplicate-key-bug/blob/master/src/pages/Index.vue#L3) to each page of the collection. When you add gridsome-plugin-i18n you get the `Error: Duplicate key for property path...`

I’d be happy to contribute a fix, though I have no clue where I’d have to start digging.

**Comment by daaru00** [comment]
Hi All,

maybe I get somethink, I think is related to template configuration and a mistake when calling [removePage](https://github.com/daaru00/gridsome-plugin-i18n/blob/master/gridsome.server.js#L62) API (passing page object instead of id). For some reason with simple pages seems works, maybe just an internal duplication check on `createPage` API.

I pushed some changes to `fix/duplicated-path` to fix it but seems it breaks the original page when calling [createPage](https://github.com/daaru00/gridsome-plugin-i18n/blob/master/gridsome.server.js#L63) API. 

I try to solve the issue, maybe I've to find an alternative solution to set the default language. Any help is welcome :smile: 

Thank you @borisdiakur for the reproduction repository!

**Comment by pmiguelmartins** [comment]
Hi Any news about this problem ???

Thanks

**Comment by daaru00** [comment]
>Hi Any news about this problem ???

@pmiguelmartins not yet !!! 

Feel free to open a PR if urgent :slightly_smiling_face: please do not spam on issue, add a +1 reaction instead.

**Comment by inakie** [comment]
Hi @daaru00 , I see you made some changes to `fix/duplicated-path` , do they solve the duplicated key issue? Thanks!

**Comment by daaru00** [comment]
Hi @inakie,

yes made some changes that currently fix this duplication error. Unfortunately a related error appears and the template pages created by the plugin are broken (seems components is not passed correctly).

I could also close this partial fix but however you would have a problem using the plugin.. perhaps with the page modification hook (is in progress by the core) will allow a more agile modification of the pages and should solve also the second related issue.

**Comment by daaru00** [comment]
Hi all,

I try to change the creating approach using pages and routes hooks. Maybe I found where the main issue comes from: I missed the `queryVariables` parameters when cloning pages. This parameters is responsible to pass data into template's page.

Now this error should disappear and template page start working as expected, can you plase do a try with the latest changes from `fix/duplicated-path` branch?

**Comment by shumiyao** [comment]
@daaru00  Thank you!!!! It is just working great here without the duplicate key for property path error!

in case anyone that wants to try `fix/duplicated-path` right away:

```
npm i daaru00/gridsome-plugin-i18n#fix/duplicated-path --save
```

**Comment by rodrigomata** [comment]
Followed @shumiyao and can confirm it's working now

**Comment by m3hari** [comment]
Confirmed it's working! @daaru00 What is left to release this?

**Comment by daaru00** [comment]
Hi @m3hari,

just doing some tests in order to prevent some regression issue. With Gridsome there are many different configurations, approaches, plugins and it's not so easy to juggle it's internal logic (route / pages / dynamic pages / managed pages).. I have to ask you for a little patience..

PS: I've created a template for this plugin into [CodeSandbox](https://codesandbox.io/s/gridsome-plugin-i18n-lsh8v) in order to test different scenarios and help to find regressions and replicate bugs. This until I can set up some functional tests :smile:

**Comment by shumiyao** [comment]
@daaru00 

When are you going to merge this fix to the master? I have been trying this for a month, it is working fine.

**Comment by cfecherolle** [comment]
This branch also fixed the issue for me! Should we merge it @daaru00 or do you have some more tests you would like to run first? If so, can we help? :)

**Comment by daaru00** [comment]
Hi all, 

sorry for the delay.. just merged in master right now and published the 1.5.0 version with above discussed fixes. 

In order to upgrade to the latest version remove the branch specification in package.json:
```json
"gridsome-plugin-i18n": "^1.5.0",
```
then upgrade the package:
```bash
npm upgrade gridsome-plugin-i18n
```
or if using yarn:
```bash
yarn upgrade gridsome-plugin-i18n
```
